sunpy/util/metadata.py

```diff
@@ -34,10 +34,12 @@ def __init__(self, *args):
             adict = args[0]
             if isinstance(adict, list) or isinstance(adict, tuple):
                 items = adict
-            elif isinstance(adict, dict):
-                items = adict.items()
             else:
-                raise TypeError(f"Can not create a MetaDict from this input of type {type(adict)}")
+                try:
+                    items = dict(adict).items()
```

Comment:
Why does the type check for the dict not wokr?

Reply:
I've also just tried this (i.e. `dict(adict).items()` from the example before from #5043  and it doesn't work, but `dict(adict)` does

Reply:
I think classes can implement special methods that tell python how to convert them to `dict`, without themselves being a child class of `dict`.